WalletWasabi/Blockchain/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -157,6 +157,25 @@ private ProcessedResult ProcessNoLock(SmartTransaction tx)
 					}
 					else // new confirmation always enjoys priority
 					{
+						var unconfirmedDoubleSpentTxId = doubleSpends.First().TransactionId;
+						if (TransactionStore.MempoolStore.TryGetTransaction(unconfirmedDoubleSpentTxId, out var stx))
+						{
+							if (stx.IsReplacement)
+							{
+								var (replaced, restored) = Coins.Undo(unconfirmedDoubleSpentTxId);
+
+								result.ReplacedCoins.AddRange(replaced);
+								result.RestoredCoins.AddRange(restored);
+
+								foreach (var replacedTransactionId in replaced.Select(coin => coin.TransactionId))
+								{
+									TransactionStore.MempoolStore.TryRemove(replacedTransactionId, out _);
+								}
+								goto here;
```

Comment:
![image](https://user-images.githubusercontent.com/9156103/110430255-614b7f00-80ac-11eb-9dd4-3a15fed16bf2.png)

Reply:
![image](https://user-images.githubusercontent.com/9156103/110430809-331a6f00-80ad-11eb-937e-2f1305b5e448.png)

Reply:
![image](https://user-images.githubusercontent.com/9156103/110430887-52190100-80ad-11eb-837c-de48c59cf2e4.png)

Reply:
- Add a comment about why are we goto.